Model/ProductData.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class ProductData
+{
+    /**
+     * Product parent types
+     */
+    const PARENT_TYPES = [\Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE, \Magento\Bundle\Model\Product\Type::TYPE_CODE, \Magento\GroupedProduct\Model\Product\Type\Grouped::TYPE_CODE];
+
+    public function __construct(
+        \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productCollection,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Bundle\Model\Product\Type $bundleType,
+        \Magento\GroupedProduct\Model\Product\Type\Grouped $groupedType,
+        \Magento\ConfigurableProduct\Model\Product\Type\Configurable $configurableType
+    )
+    {
+        $this->productCollection = $productCollection;
+        $this->storeManager = $storeManager;
+        $this->bundleType = $bundleType;
+        $this->groupedType = $groupedType;
+        $this->configurableType = $configurableType;
+    }
+
+    public function getProducts($storeId)
+    {
+        $products = $this->productCollection->create()->addAttributeToSelect('*')->addUrlRewrite()->addStoreFilter($storeId);
+
+        foreach ($products as $product) {
+            $productId = $product->getId();
+            $productType = $product->getTypeId();
+
+            if ($productType == "simple" || $productType == "virtual") {
+                if ($this->getParentId($productId, $productType)) {
+                    continue;
+                }
+            }
+
+            $productsArray[] = [
+                'categories' => $product->getCategoryIds(),
+                'id'         => $productId,
+                'sku'        => $product->getSku(),
+                'imageUrl'   => (!empty($product->getImage())) ? $this->getProductImageUrl($product->getImage()) : '',
```

Comment:
lets move all these conditional out of the hash initialization to make it easier to read